app/MMB/src/scenes/tripOverview/TimelineTitle.js

```diff
@@ -29,38 +30,25 @@ type PropsWithContext = {|
   +warnings: ?(Warning[]),
 |};
 
-type State = {|
-  warning: ?Warning,
-|};
-
-class TimelineTitle extends React.Component<PropsWithContext, State> {
-  constructor(props) {
-    super(props);
-    this.state = {
-      warning: null,
-    };
-  }
-  componentDidUpdate(prevProps) {
-    const { warnings } = this.props;
-    if (prevProps.warnings && warnings) {
-      if (prevProps.warnings.length !== warnings.length) {
-        const data = this.props.data;
-        const localTime = idx(data, _ => _.localTime);
-        const iataCode = idx(data, _ => _.airport.locationId);
-
-        const warning = warnings.find(
-          _warning =>
-            _warning.timelineTitle.localTime == localTime &&
-            _warning.timelineTitle.iataCode == iataCode,
-        );
-        this.setState({ warning });
-      }
-    }
-  }
+class TimelineTitle extends React.Component<PropsWithContext> {
+  getWarning = memoize(
```

Comment:
This is to have it executed only once, right?

Reply:
`memoize-one simply remembers the last arguments, and if the function is next called with the same arguments then it returns the previous result.`

Reply:
I see, to optimize calls. Cool.